packages/babel-plugin/src/index.tsx

```diff
@@ -13,39 +14,52 @@ export default declare<State>((api) => {
     visitor: {
       Program: {
         exit(path, state) {
-          if (state.compiledImportFound && !path.scope.getBinding('React')) {
+          if (state.compiledImports && !path.scope.getBinding('React')) {
             // React is missing - add it in at the last moment!
             path.unshiftContainer('body', template.ast(`import * as React from 'react'`));
           }
         },
       },
       ImportDeclaration(path, state) {
         if (path.node.source.value === '@compiled/core') {
-          state.compiledImportFound = true;
+          // The presence of the module enables CSS prop
+          state.compiledImports = {};
+
           path.node.specifiers = path.node.specifiers
-            .filter(
-              (specifier) =>
-                specifier.local.name !== 'styled' && specifier.local.name !== 'ClassNames'
-            )
+            .filter((specifier) => {
+              if (!state.compiledImports || !t.isImportSpecifier(specifier)) {
+                return false;
+              }
+
+              if (specifier.imported.name === 'styled') {
+                state.compiledImports.styled = specifier.local.name;
+                // Remove the import
+                return false;
+              }
+
+              // Keep the import
+              return true;
+            })
+            // Add on the util imports we're going to use later in the transform.
             .concat([importSpecifier('CC'), importSpecifier('CS')]);
         }
       },
       TaggedTemplateExpression(path, state) {
-        if (!state.compiledImportFound) {
+        if (!state.compiledImports?.styled) {
```

Comment:
Optional chaining is dope. I remember from coffescript days how we used to do this. It was awesome. This is the one I am talking about http://coffeescript.org/#existential-operator